scripts/helper.sh

```diff
@@ -38,3 +38,67 @@ function fail_if_error() {
     exit $retVal
   fi
 }
+
+function check_image_artifact_exists() {
+  local FILE=$1
+  if [[ ! -f "$FILE" ]]; then
+    echo "Error: Image artifact $FILE does not exist."
+    set_job_status "error"
+    exit 1
+  fi
+}
+
+# Set up the resolv.conf file for the image root
+# This is used to ensure that the image root can resolve DNS names during customization.
+function setup_resolv() {
+    # Copy the container's /etc/resolv.conf to the image root for customization
+    local IMAGE_ROOT_DIR=$1
+    local IMAGE_ROOT_RESOLV=$IMAGE_ROOT_DIR/etc/resolv.conf
+    local TMP_RESOLV=$IMAGE_ROOT_DIR/tmp/resolv.conf
+    if [ -f "$IMAGE_ROOT_RESOLV" -o -L "$IMAGE_ROOT_RESOLV" ]; then
+        mv "$IMAGE_ROOT_RESOLV" "$TMP_RESOLV"
+    fi
+    cp --remove-destination /etc/resolv.conf "$IMAGE_ROOT_RESOLV"
+}
+
+# Restore the image root's old /etc/resolv.conf after customization
+function restore_resolv() {
+    # Restore the image root's old /etc/resolv.conf after customization
+    local IMAGE_ROOT_DIR=$1
+    local IMAGE_ROOT_RESOLV=$IMAGE_ROOT_DIR/etc/resolv.conf
+    local TMP_RESOLV=$IMAGE_ROOT_DIR/tmp/resolv.conf
+    rm "$IMAGE_ROOT_RESOLV"
+    if [ -f "$TMP_RESOLV" -o -L "$TMP_RESOLV" ]; then
+        mv "$TMP_RESOLV" "$IMAGE_ROOT_RESOLV"
+    fi
+}
+
+# Fix permissions on /image dir if it exists
+function check_image_dir() {
+    # Change ownership and permissions on /image dir if it exists.
+    #  This dir contains config files that may have sensitive information
+    #  in them and should only be readable by root.
+    local IMAGE_ROOT_DIR=$1
+    if [[ -d ${IMAGE_ROOT_DIR}/image ]]; then
+      # change read/write permissions of dir to root only
+      chown root ${IMAGE_ROOT_DIR}/image
+      chmod 700 ${IMAGE_ROOT_DIR}/image
+
+      # change files in the dir to root ownership and only rw for root
+      chown root ${IMAGE_ROOT_DIR}/image/*
+      chmod 600 ${IMAGE_ROOT_DIR}/image/*
+
+      # change the .sh file to rwx for root
+      chmod 700 ${IMAGE_ROOT_DIR}/image/*.sh
+    fi
+}
+
+# Wait for a file to be created to proceed
+function wait_for_file() {
+    # Wait for the user to signal they are done
+    echo "Waiting for $1 to be created"
+    until [ -f "$1" ]
```

Comment:
On the off chance that the used created a directory in that location with that name, it seems like we'd still want to interpret that as a desire to exit. Obviously not something that we'd expect to happen, so not a big deal either way.
```suggestion
    until [ -e "$1" ]
```

Reply:
Done.